api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,252 @@
+            // no need to log these errors, just throw them back
+            logger.trace("Exiting getVehicle");
+            throw e;
+        } catch (Exception e) {
+            // log all unhandled exceptions and throw an internal server error
+            logger.error(e);
+            logger.trace("Exiting getVehicle");
+            throw new InternalServerErrorException(e, awsRequestId);
+        }
+    }
+
+    private Optional<MotrResponse> getMotVehicle(String registration) throws Exception {
+        logger.info(String.format("Retrieving vehicle by registration: %s", registration));
+
+        MotrResponse motVehicles;
+
+        try {
+            motVehicles = motrReadService.getLatestMotTestByRegistration(registration, false);
+        } catch (Exception e) {
+            logger.error(String.format("There was an error retrieving mot vehicle with reg: %s",
+                    registration), e);
+            throw new InvalidResourceException("There was an error retrieving mot vehicle", awsRequestId);
+        }
+
+        return Optional.ofNullable(motVehicles);
+    }
+
+
+
+    private Optional<DvlaVehicle> getDvlaVehicle(String registration) throws Exception {
+        logger.info(String.format("Retrieving dvla vehicle by registration: %s", registration));
+
+        DvlaVehicle dvlaVehicle;
+
+        try {
+            dvlaVehicle = vehicleReadService.getDvlaVehicleByRegistrationWithVin(registration);
+        } catch (Exception e) {
+            logger.error(String.format("There was an error retrieving dvla vehicle with reg: %s",
+                    registration), e);
+            throw new InvalidResourceException("There was an error retrieving dvla vehicle", awsRequestId);
+        }
+
+        return Optional.ofNullable(dvlaVehicle);
+    }
+
+    private Optional<Vehicle> getHgvPsvVehicle(String registration) throws Exception {
+        Vehicle foundVehicle = null;
+
+        try {
+            if (!Strings.isNullOrEmpty(registration)) {
+                foundVehicle = hgvVehicleProvider.getVehicle(registration);
+            } else {
+                logger.warn("Passed registration was empty or null");
+            }
+        } catch (Exception e) {
+            logger.error("There was an error retrieving the HGV/PSV history", e);
+            throw new InvalidResourceException("There was an error retrieving the HGV/PSV history", awsRequestId);
+        }
+
+        return Optional.ofNullable(foundVehicle);
+    }
+
+    private HgvPsvVehicle setResponseHgvVehicle(Vehicle vehicle, DvlaVehicle dvlaVehicle) throws Exception {
+        HgvPsvVehicle hgvPsvVehicle = new HgvPsvVehicle();
+
+        hgvPsvVehicle.setMake(vehicle.getMake());
+        hgvPsvVehicle.setModel(vehicle.getModel());
+        hgvPsvVehicle.setPrimaryColour(dvlaVehicle.getColour1().getName());
+        hgvPsvVehicle.setRegistration(dvlaVehicle.getRegistration());
+        hgvPsvVehicle.setVin(dvlaVehicle.getVin());
```

Comment:
We don't use vin anymore, this should be removed (in other places where we added this as well)